File: 3142-check-if-grid-satisfies-conditions/3142-check-if-grid-satisfies-conditions.cpp
class Solution {
public:
    bool satisfiesConditions(vector<vector<int>>& grid) {
        for(int col = 0; col < grid[0].size(); col++) {
            for(int row = 1; row < grid.size(); row++) {
                if(grid[row][col] != grid[0][col]) return false;
            }
        }
        for(int col = 1; col < grid[0].size(); col++) {
            if(grid[0][col] == grid[0][col - 1]) return false;
        }
        return true;
    }
};

File: 3142-check-if-grid-satisfies-conditions/3142-check-if-grid-satisfies-conditions.py
class Solution:
    def satisfiesConditions(self, grid: List[List[int]]) -> bool:
        col=[]
        flag_for_col=True
        flag_for_row=True

        for r in range(len(grid)):
            for k in range(len(grid[0])-1):
                if (grid[r][k]==grid[r][k+1]):
                    flag_for_row=False


        for i in range(len(grid[0])):
            cols=[]
            for j in range(len(grid)):
                cols.append(grid[j][i])
            if len(set(cols))!=1:
                flag_for_col=False
            col.append(cols)
            
        if flag_for_row and flag_for_col:
            return (True)
        return (False)